include/librealsense2/h/rs_device.h

```diff
@@ -353,6 +353,151 @@ rs2_raw_data_buffer* rs2_serialize_json(rs2_device* dev, rs2_error** error);
+
+/**
+* \brief Gets RealSense firmware log message data.
+* \param[in] msg	    firmware log message object
+* \param[out] error     If non-null, receives any error that occurs during this call, otherwise, errors are ignored.
+* \return               pointer to start of the firmware log message data
+*/
+const unsigned char* rs2_firmware_log_message_data(rs2_firmware_log_message* msg, rs2_error** error);
+
+/**
+* \brief Gets RealSense firmware log message size.
+* \param[in] msg	    firmware log message object
+* \param[out] error     If non-null, receives any error that occurs during this call, otherwise, errors are ignored.
+* \return               size of the firmware log message data
+*/
+int rs2_firmware_log_message_size(rs2_firmware_log_message* msg, rs2_error** error);
+
+
+/**
+* \brief Gets RealSense firmware log message timestamp.
+* \param[in] msg	    firmware log message object
+* \param[out] error     If non-null, receives any error that occurs during this call, otherwise, errors are ignored.
+* \return               timestamp of the firmware log message
+*/
+unsigned int rs2_firmware_log_message_timestamp(rs2_firmware_log_message* msg, rs2_error** error);
+
+/**
+* \brief Gets RealSense firmware log message severity.
+* \param[in] msg	    firmware log message object
+* \param[out] error     If non-null, receives any error that occurs during this call, otherwise, errors are ignored.
+* \return               severity of the firmware log message data
+*/
+rs2_log_severity rs2_firmware_log_message_severity(const rs2_firmware_log_message* msg, rs2_error** error);
+
+/**
+* \brief Creates RealSense firmware logs parser.
+* \param[in] xml_path	path to xml file needed for parsing
+* \param[out] error  If non-null, receives any error that occurs during this call, otherwise, errors are ignored.
+* \return            firmware logs parser object
+*/
+rs2_firmware_log_parser* rs2_create_firmware_log_parser(const char* xml_path, rs2_error** error);
+
+/**
+* \brief Frees the relevant firmware logs parser object.
+* \param[in] firmware logs parser object that is no longer needed
+*/
+void rs2_delete_firmware_log_parser(rs2_firmware_log_parser* parser);
+
+/**
+* \brief Gets RealSense firmware log parser
+* \param[in] fw_logs_parser	    firmware log parser
+* \param[in] fw_log_msg	        firmware log message to be parsed
+* \param[out] error             If non-null, receives any error that occurs during this call, otherwise, errors are ignored.
+* \return                       firmware log parsed message
+*/
+rs2_firmware_log_parsed_message* rs2_parse_firmware_log(rs2_firmware_log_parser* fw_logs_parser,
+    rs2_firmware_log_message* fw_log_msg, rs2_error** error);
+
+/**
+* Delete RealSense firmware log parsed message
+* \param[in]  device    Realsense firmware log parsed message to delete
+*/
+void rs2_delete_firmware_log_parsed_message(rs2_firmware_log_parsed_message* fw_log_parsed_msg);
+
```

Comment:
There seem to be enough APIs to refactor it into its own header file, what do you think?